Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Magic Spells *Fire Lifestyle*

https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/899
reattached
but if someone gonna work on that firestyle branch you need to update that branch to current magic2024
## Changelog:
- Added Elemental fire magic spells and rituals
- Created spellcasting GUI, associated events, and the ability to cast spells


## Developer changelog:
- Added this thing.
- Changed that thing.


## Tests:
- [x] Spell tooltips are correct.
- [ ] Spell effects work as intended
- [ ] Event effects / loc work as intended
- [ ] Minimal impact on game performance (if possible)

Issues from Testers assigned to @helloruhe & @robmart:
- [x] Spell casting complete loc weird
- [x] County/Development tooltip updates
- [x] Spells targeting multiple titles can be cast without a target (might be fixed with list clear update)
- [x] Fix backdraft rank 2/3 not working/showing
- [x] Make sure backdraft consumes correctly
- [x] Clear list when selecting a single target spell
- [x] Clear list when selecting a new spell or rank
- [x] ~~Spell rank GUI dropdown bugged **UNABLE TO CONSISTENTLY REPLICATE**~~ Resolved with GUI overhaul
- [x] Choose title button has a weird active area
- [x] Spell rank game concept not showing tooltip
- [x] Minor localization changes
- [x] Fix ritual CD times
- [x] ~~Restrict focus selection~~ Moved to magic-2024
- [x] Fix Spirit/Decay trait assignment 
- [x] Fix mana resource disappearing
- [x] Vernal Solstice
- [x] ~~Gender doctrine loc~~ Moved to magic-2024
- [x] Clean error log **once everything above is completed**

# How to test:
- You can use add_elemental_fire_magic_lifestyle_xp 18000 in the debug menu to unlock all 18 perks and all spells
- You can choose a target for your spell by opening the interaction menu on your target and clicking "Cast Spell"
- Try to cast each spell on different targets at different times if possible. 
- Ensure each spell follows the appropriate cooldown and cast time timer (indicated on GUI)
- Ensure all restrictions on casting spells work, and it's clear what is preventing you from casting a spell. (E.G. if you cast a spell on someone's province you shouldn't be able to do it again if they still have the effect on the province, and the GUI should tell you that in the tooltip)
- Inspect text for spelling/grammar mistake or error text
- Ensure the spell actually follows the intended effect as outlined in the tooltip
